src/lib/devp_comms.c

```diff
@@ -505,9 +509,16 @@ static void sleep_block_comms (comms_layer_t * p_comms)
 		{
 			if(NULL != mp_sleep_ctrl[i]) // Sleep control is optional
 			{
-				debug1("block %d", i);
+				#ifdef SERVICE_MODE_TIMEOUT
+				m_rcomm_keepalive_timeout = devp_service_mode_timeout_get();
+				#endif //SERVICE_MODE_TIMEOUT
+				if(!m_rcomm_keepalive_timeout)
+				{
+					m_rcomm_keepalive_timeout = DEVP_DEFAULT_SLEEP_TIMEOUT_MS;
+				}
+				debug1("block %d for %d sec", i, m_rcomm_keepalive_timeout/1000);
 				comms_sleep_block(mp_sleep_ctrl[i]);
-				osTimerStart(m_sleep_timer, DEVP_SLEEP_TIMEOUT_MS);
+				osTimerStart(m_sleep_timer, m_rcomm_keepalive_timeout);
```

Comment:
I think if the timer is started with a tick count of 0, then it expires immediately. Is it expected behavior? Or maybe if m_rcomm_keepalive_timeout is 0, then timer should not be even started?

Reply:
timer should not be started when tick count is 0.
that's why there is check to set tick count to default if value got from devp happens to be 0.